src/renderers/shaders/ShaderChunk/dithering_pars_fragment.glsl

```diff
@@ -0,0 +1,14 @@
+// based on https://www.shadertoy.com/view/MslGR8
+vec3 dithering( vec3 color ) {
+	//Calculate grid position
+	float grid_position = fract( dot( gl_FragCoord.xy - vec2( 0.5, 0.5 ) , vec2( 1.0 / 16.0, 10.0 / 36.0 ) + 0.25 ) );
+
+	//Shift the individual colors differently, thus making it even harder to see the dithering pattern
+	vec3 dither_shift_RGB = vec3( 0.25 / 255.0, -0.25 / 255.0, 0.25 / 255.0 );
+
+	//modify shift acording to grid position.
+	dither_shift_RGB = mix( 2.0 * dither_shift_RGB, -2.0 * dither_shift_RGB, grid_position );
+
+	//shift the color by dither_shift
+	return color + 0.5 / 255.0 + dither_shift_RGB;
```

Comment:
I didn't expect is that GPUs always truncate (round down) when converting from floating point to integer output.  I figured it used a midpoint round, and thus adding 0.5 would bias it up.

If it is truncating, basically every GPU-based renderer in the world should be adding 0.5/255.0 to their shaders for every pass.  I am confused that I've never seen this before as I thought with a lot of the re-enterance shaders (processing the results of a previous frame and generating a new one) this would be noticeable as a half bit shift down per frame.

I did some google searches and I can not find anyone saying that the GPU rounds down when converting floating point to bytes, but I can not prove otherwise either.  Sort of a grey area.  What does the OpenGL specification say here?

(BTW I know about quantization -- I wrote the banding removal for the manually SSAA shader by adding randomness to here to the weights of the passes to avoid accumulating rounding/truncation errors: https://github.com/mrdoob/three.js/blob/dev/examples/js/postprocessing/SSAARenderPass.js#L116 )